### PR TITLE
Update github actions to use composite action created before

### PR DESCRIPTION
**Pull request by mohzulfikar**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by mohzulfikar** [comment]
Here, I modified the github action to use the composite action I created on Homebrew/actions#289.

**Comment by mistydemeo** [comment]
I think you're right. We should look into that. @mohzulfikar, you can see the options that are set for bash here: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsshell

**Comment by carlocab** [comment]
Wild guess: the action doesn't inherit permissions set in the calling workflow

File: .github/workflows/tests.yml
Comment (dawidd6) [inline .github/workflows/tests.yml:236]: What about this step? Shouldn't it be done inside the action too?

**Comment by mohzulfikar** [comment]
That's weird actually. It passed on macos but not on linux. Any idea on how to debug this?

**Comment by carlocab** [comment]
Probably user permissions -- the container runs as user `linuxbrew`, but GitHub Actions expects something else. It may be that it's the GHA user running your action?

**Review by MikeMcQuaid — DISMISSED** [review]

File: .github/workflows/tests.yml
Comment (carlocab) [inline .github/workflows/tests.yml:24]: Did you remove this for testing? We probably want to keep this. Same below.

Comment (mohzulfikar) [reply .github/workflows/tests.yml:24]: Sorry that, Wrong commit, I'm testing on my fork

**Comment by mohzulfikar** [comment]
Hi there, sorry for stalling this PR. I've found the root cause for failing CI on Linux. It's because the working directory is not the default (set on the matrix). I want to test this change using my modified composite action first to see if it's works.

**Comment by mohzulfikar** [comment]
But, it seems like the CI is not triggered, is there something I miss?

**Comment by MikeMcQuaid** [comment]
> mohzulfikar/actions@main is not allowed to be used in Homebrew/homebrew-core. Actions in this workflow must be: within a repository owned by Homebrew, created by GitHub, or matching the following: Vampire/setup-wsl@*, codecov/codecov-action@*, google-github-actions/setup-gcloud@*, peter-evans/*, reitermarkus/*, ruby/setup-ruby@*, dessant/lock-threads@*, google-github-actions/auth@*, google-github-actions/get-gke-credentials@*.

**Comment by mohzulfikar** [comment]
I see, thanks for the confirmation. Seems like I need to modify the composite action on the Homebrew actions repository.

**Comment by mohzulfikar** [comment]
It seems like all checks have been successful. Any more feedback for this?

**Review by MikeMcQuaid — APPROVED** [review]

**Review by mistydemeo — APPROVED** [review]

**Comment by mistydemeo** [comment]
It looks like we're successful here. I'll go ahead and merge, and we can watch it in regular use.

File: .github/workflows/tests.yml
Comment (mistydemeo) [reply .github/workflows/tests.yml:238]: Looking back at this comment - let me know what your plans are for this after we merge this.

File: .github/workflows/dispatch-build-bottle.yml
Comment (carlocab) [inline .github/workflows/dispatch-build-bottle.yml:135]: Not a fan of the way this combines the failures summary and `brew bottle` output into one step:

https://github.com/Homebrew/homebrew-core/runs/7869970862?check_suite_focus=true#step:7:1

I'm often interested in only one of those at any given point, and I'd rather only see what I'm looking for when I expand an item in the workflow output. (I imagine other maintainers have similar opinions, but feel free to point out if I'm mistaken here.)

Is there any way we can split these back up?

Comment (MikeMcQuaid) [reply .github/workflows/dispatch-build-bottle.yml:135]: @carlocab if this is worse for you: feel free to revert this.

Comment (mohzulfikar) [reply .github/workflows/dispatch-build-bottle.yml:135]: One option we could implement is to instead group `brew test-bot' and its following result. I thought `steps_output.txt` and `bottle_output.txt` were products of the same `brew test-bot` steps so it's okay to group them.

Comment (carlocab) [reply .github/workflows/dispatch-build-bottle.yml:135]: `steps_output.txt` is just a duplicate of the tail of the step that preceded it (the step with `id: brew-test-bot-formulae`). To see this, compare the [tail of `brew-test-bot-formulae`](https://github.com/Homebrew/homebrew-core/runs/7869970862?check_suite_focus=true#step:6:5312) with the [contents of `steps_output.txt`](https://github.com/Homebrew/homebrew-core/runs/7869970862?check_suite_focus=true#step:7:27).

Dumping it anyway is useful because it gives you a way of viewing that tail without a lot of extraneous information. Putting it with other things defeats much of the purpose of creating `steps_output.txt`.

> One option we could implement is to instead group `brew test-bot` and its following result.

Could you elaborate on what you had in mind?

Comment (carlocab) [reply .github/workflows/dispatch-build-bottle.yml:135]: I've opened https://github.com/Homebrew/actions/pull/295.

**Pull request merged**
